var/spack/repos/builtin/packages/gcc/package.py

```diff
@@ -266,6 +274,9 @@ class Gcc(AutotoolsPackage, GNUMirrorPackage):
         patch('darwin/gcc-4.9.patch1', when='@4.9.0:4.9.3')
         patch('darwin/gcc-4.9.patch2', when='@4.9.0:4.9.3')
 
+    # https://go-review.googlesource.com/c/gofrontend/+/344189 fixes ICE in gccgo
+    patch('go1.17-bootstrap.patch', when='@10:11.2')
```

Comment:
I guess this should be `when='@10.0:10.3,11.0:11.2'`?

Also, fetching the patch would allow to not carry the patch in the repository: https://gcc.gnu.org/git/?p=gcc.git;a=patch;h=7edb992d8195572e2c823fedab319eb6cb2f0f32

Reply:
Done.
```diff
-    patch('go1.17-bootstrap.patch', when='@10:11.2')
+    patch('https://gcc.gnu.org/git/?p=gcc.git;a=patch;h=7edb992d8195572e2c823fedab319eb6cb2f0f32',
+          sha256='97e7b46f385d8e47545cd28390aaeed3a5186b48d78507ba2641a19a0de98504',
+          when='@10.0:10.3,11.0:11.2')
```